packages/shopware-6-client/src/services/cartService.ts

```diff
@@ -0,0 +1,74 @@
+import { Cart } from "../interfaces/models/checkout/cart/Cart";
+import {
+  getCheckoutCartEndpoint,
+  getCheckoutCartProductEndpoint,
+  getCheckoutPromotionCodeEndpoint,
+  getCheckoutCartLineItemEndpoint
+} from "../endpoints";
+import { apiService } from "../apiService";
+import { ContextTokenResponse } from "../interfaces/response/ContextTokenResponse";
+
+export async function createCart(): Promise<ContextTokenResponse> {
+  const resp = await apiService.post(getCheckoutCartEndpoint());
+  let contextToken = resp.data["sw-context-token"];
+  return { contextToken };
+}
+
+export async function getCart(): Promise<Cart> {
+  const resp = await apiService.get(getCheckoutCartEndpoint());
+
+  return resp.data;
+}
+
+export async function addProduct(
+  productId: string,
+  newQuantity: number
+): Promise<Cart> {
+  let params = { quantity: newQuantity };
+  const resp = await apiService.post(
+    getCheckoutCartProductEndpoint(productId),
+    params
+  );
+
+  return resp.data;
+}
+
+export async function addLineItem(
+  itemId: string,
+  addQuantity: number
+): Promise<Cart> {
+  let params = { type: "product", quantity: addQuantity };
+  const resp = await apiService.post(
+    getCheckoutCartLineItemEndpoint(itemId),
+    params
+  );
+
+  return resp.data;
+}
+
+export async function updateLineItem(
+  itemId: string,
+  newQuantity: number
+): Promise<Cart> {
+  let params = { quantity: newQuantity };
+  const resp = await apiService.patch(
+    getCheckoutCartLineItemEndpoint(itemId),
+    params
+  );
+
+  return resp.data;
+}
+
+export async function removeLineItem(itemId: string): Promise<Cart> {
+  const resp = await apiService.delete(getCheckoutCartLineItemEndpoint(itemId));
+
+  return resp.data;
+}
+
+export async function addCode(promotionCode: string): Promise<Cart> {
```

Comment:
just for method to be more searchable
```suggestion
export async function addPromotionCode(promotionCode: string): Promise<Cart> {
```

Reply:
also - don't we need option `removePromotionCode`? It's pretty common to have it.
Especially when we have multiple codes, but even with one. (Can we have multiple codes?)

Reply:
Name of the method changes.

For removePromotionCode we may use removeCartLineItem. I will write a JSDoc for that.